rascontrol/rascontrol.py

```diff
@@ -232,62 +277,191 @@ def __init__(self, version='41'):
 
-    def simple_xs_list(self): 
+    def simple_xs_list(self):
         """
         Returns list of XS as SimpleXS objects, all names are strip()ed
 
         This is primarily for interacting with parserasgeo
         """
-        # Check for xs list, create if necessary
-        if self.xs_list is None:
+        simple_list = self._simple_node_list('xs')
+        return simple_list 
+
+    def get_xs(self, xs_id, river = None, reach = None):
+        """
+        Returns requested cross-sections, ignores river and reach if not specified
+        Raises CrossSectionNotFound if station is not in model
+        
+        :param xs_id: id of station (cast to string automatically)
+        :param river: river name (string)
+        :param reach: reach name (string)
+        :return: Node object for station
+        """
+        node = self._get_node('xs', xs_id, river, reach)
+        return node
+
+    def simple_culvert_list(self):
+        """
+        Returns list of culverts as SimpleCulvert objects, all names are strip()ed
+
+        This is primarily for interacting with parserasgeo
+        """
+        simple_list = self._simple_node_list('culvert')
+        return simple_list 
+
+    def get_culvert(self, culvert_id, river = None, reach = None):
+        """
+        Returns requested cross-sections, ignores river and reach if not specified
+        Raises CrossSectionNotFound if station is not in model
+        
+        :param culvert_id: id of station (cast to string automatically)
+        :param river: river name (string)
+        :param reach: reach name (string)
+        :return: Node object for station
+        """
+        node = self._get_node('culvert', culvert_id, river, reach)
+        return node        
+    
+    # - - - - - - - - - - - - -
+    # semi-private methods for get_<node> and simple_<node>_list
+    # - - - - - - - - - - - - -
+
+    def _simple_node_list(self, node_type): 
+        """
+        Returns list of station as Simple<stataion> objects, all names are strip()ed
+
+        This is primarily for interacting with parserasgeo
+        """
+
+        if node_type == 'xs':
+            node_list = self.xs_list
+        elif node_type == 'culvert':
+            node_list = self.culvert_list
+        elif node_type == 'bridge':
+            node_list = self.bridge_list
+        elif node_type == 'mult_open':
+            node_list = self.mult_open_list
+        elif node_type == 'inline_struct':
+            node_list = self.inline_struct_list
+        elif node_type == 'lateral_struct':
+            node_list = self.lateral_struct_list
+
+        # Check for node list, create if necessary
+        if node_list is None:
             if not self.project_is_open:
                 raise NoProject('Project must be opened before calling RasController.get_xs()')
-            self.xs_list = self._load_xs_list()
+            node_list = self._load_node_list(node_type)
```

Comment:
Is this line intended to populate self.xs_list, or self.culvert_list, etc? It appears to reassign node_list to refer to the list returned by _load_node_list(), versus actually updating self.xs_list, etc.